web/quiz_views.py

```
@@ -297,37 +488,75 @@ def take_quiz(request, quiz_id):
+        if not incomplete_attempt:
+            user_quiz = UserQuiz(quiz=quiz, user=user)
+            user_quiz.save()
+            request.session["active_quiz_id"] = user_quiz.id
+            request.session.save()
+        else:
+            user_quiz = incomplete_attempt
 
-    user_quiz = UserQuiz(quiz=quiz, user=user)
-    user_quiz.save()
+        # Now define the helper AFTER user_quiz exists
+        def calculate_remaining_time():
```

Ruff (0.8.2)

Missing return type annotation for private function `calculate_remaining_time`

(ANN202)